src/components/serviceOfferings/ServiceOfferings.tsx

```diff
@@ -1,193 +1,49 @@
-      authContext,
-      targetClass
-    );
-    const map = mapServiceOfferings(selfDescriptions.data);
-    console.log('Map:', map);
-    setSelfDescriptionData(map);
-    setIsLoading(false);
+  const search = (filter: string) => {
+    setSearchText(filter)
   }
 
   return (
-    <div className="service-offerings">
-      <header>
-        <h2>Service Offerings</h2>
-      </header>
-      {authContext.isAuthenticated && (
-        <div className="content">
-          <FormControl sx={{ minWidth: 200 }}>
-            <InputLabel id="shape-label">Select Shape</InputLabel>
-            <Select
-              labelId="shape-label"
-              id="shape-select"
-              value={selectedShape.shortSubject}
-              label="SHACL Shape"
-              onChange={handleShapeChange}
-            >
-              {shapes.map((shape) => (
-                <MenuItem key={shape.shortSubject} value={shape.shortSubject}>
-                  {shape.shortSubject}
-                </MenuItem>
-              ))}
-            </Select>
-          </FormControl>
-          {isShapeSelected && (
-            <FormControl sx={{ marginX: 1, minWidth: 200 }}>
-              <InputLabel id="property-label">Property</InputLabel>
-              <Select
-                labelId="property-label"
-                id="property-select"
-                value={selectedProperty}
-                label="Property"
-                onChange={handlePropertyChange}
-              >
-                {properties.map((property) => (
-                  <MenuItem key={property} value={property}>
-                    {property}
-                  </MenuItem>
-                ))}
-              </Select>
-            </FormControl>
-          )}
-          {isPropertySelected && (
-            <FormControl sx={{ minWidth: 200 }}>
-              <TextField
-                id="outlined-basic"
-                label="Keyword"
-                variant="outlined"
-              />
-            </FormControl>
-          )}
-          <div className="button">
-            {isShapeSelected && (
-              <Button variant="contained" onClick={handleSearch} size="large">
-                Search
-              </Button>
-            )}
-          </div>
-
-          <Padding key="i01" paddingTop="20px" />
-          <div>
-            {!isLoading &&
-              selfDescriptionData.length > 0 &&
-              selfDescriptionData.map((selfDescription) => {
-                return (
+    <>
+      <Header title={`${t('service-offerings.titles')} (${serviceOfferings.length} ${t('common.results')})`}/>
+      <Main>
+        <LoadingIndicator isLoading={state === 'LOADING'}/>
+        <SearchBar placeholder={t('resources.search-bar-text')} onSearch={search}/>
+        <CardContainer isLoaded={state === 'LOADED'}>
+          {
+            serviceOfferings
+              .filter(selfDescription => JSON.stringify(selfDescription).includes(searchText))
```

Comment:
Ich habe die Suche ähnlich implementiert. Doch ist es nicht eigentlich schöner, über die Elemente und deren variablen zu iterieren anstatt JSON.stringify zu nutzen?

Reply:
I have implemented the search in a similar way. But isn't it actually nicer to iterate over the elements and their variables instead of using JSON.stringify?